icu4c/source/i18n/uspoof.cpp

```diff
@@ -697,6 +801,64 @@ uspoof_getSkeleton(const USpoofChecker *sc,
     return destStr.length();
 }
 
+U_CAPI int32_t U_EXPORT2 uspoof_getBidiSkeleton(const USpoofChecker *sc, UBiDiDirection direction,
+                                                const UChar *id, int32_t length, UChar *dest,
+                                                int32_t destCapacity, UErrorCode *status) {
+
+    SpoofImpl::validateThis(sc, *status);
+    if (U_FAILURE(*status)) {
+        return 0;
+    }
+    if (length < -1 || destCapacity < 0 || (destCapacity == 0 && dest != nullptr)) {
```

Comment:
This check is wrong. Contradict with your API design. Your API doc states
"@param length  .... The length of the input identifier,...., or -1 if the string is zero terminated."
"@param destCapacity .... The destCapacity may be zero... "
You neither need to check the "destCapacity < 0" nor "dest != nullptr" and could just pass to destStr.extract() to let that function to check that. 

The whole function should be implemented as simple as
```
UnicodeString destStr;
uspoof_getBidiSkeletonUnicodeString(sc, direction, UnicodeString((length == -1), id, length), destStr, status);
return destStr.extract(dest, destCapacity, *status);
```

Reply:
Removed the `destStr` checks, leaving them to `extract`. Kept the check for invalid `length` (via `isBogus`), as above.